### PR TITLE
Delete AutoPropertyMixin

### LINKED ISSUE #18355: Remove AutoPropertyMixin
## Bug Description

AutoPropertyMixin python class doesn't work with multiprocessing in python 3.9 on macos due to changes in how processes are created.

The class is only used in one place and similar features can be gained using the builtin "dataclasses" added to python 3.7.

## Steps to Reproduce

Run python/mooseutils on macos with python 3.9.

## Impact

Preventing tests from passing on CIVET.

### PR DESCRIPTION
**Pull request by aeslaughter**
Big surprise, this was more difficult than I thought. I don't think this is done.

(closes #18355)

### CODE REVIEW

**Comment by aeslaughter** [comment]
@dschwen This is ready now.

File: python/MooseDocs/base/executioners.py [hunk 1]
```diff
@@ -434,21 +405,27 @@ class ParallelBarrier(Executioner):
     def execute(self, nodes, num_threads=1):
         """Perform the translation with multiprocessing."""
 
-        barrier = multiprocessing.Barrier(num_threads)
-        manager = multiprocessing.Manager()
+        # The method for spawning processes changed to "spawn" for macOS in python 3.9, currently
+        # MooseDocs does npt work with this combination. This will be remedied in moosetools
+        # migration.
+        if (platform.python_version() >= '3.9') and (platform.system() == 'Darwin'):
+            ctx = multiprocessing.get_context('fork')
```
Comment (dschwen) [inline python/MooseDocs/base/executioners.py:412]: why is this Darwin specific? googling around I see that people seem to choose different methods (`spawn` over `fork` for macOS) but shouldn't the use of get_context method should only depend on the python version?

Comment (dschwen) [reply python/MooseDocs/base/executioners.py:412]: I'm confused. If the OS is not Darwin, it presumably is Linux, and there `fork` is the default. On mac `spawn` is the default method. So with >=3.9 on macs you're choosing fork over spawn?

**Review by dschwen — APPROVED** [review]
This needs to get it ASAP. I'll ask you tomorrow about that multiprocessing detail.

**Comment by dschwen** [comment]
Failures are Mastodon related.

**Pull request merged**